src/web/mdns-publisher/mdns_publisher.hpp

```diff
@@ -54,33 +56,60 @@ namespace Mdns {
 class Publisher
 {
 public:
-    static int StartServer(void);
-    static void SetServiceName(const char *aServiceName);
-    static void SetNetworkNameTxT(const char *aTxtData);
-    static void SetEPANIDTxT(const char *aTxtData);
-    static void SetType(const char *aType);
-    static void SetPort(uint16_t aPort);
-    static int UpdateService(void);
-    static bool IsServerStart(void) { return isStart; };
+    int StartServer(void);
+    int StartClient(void);
+    void SetServiceName(const char *aServiceName);
+    void SetNetworkNameTxT(const char *aTxtData);
+    void SetEPANIDTxT(const char *aTxtData);
+    void SetType(const char *aType);
+    void SetPort(uint16_t aPort);
+    int UpdateService(void);
+
+    int CreateService(AvahiServer *aServer);
+    int CreateService(AvahiClient *aClient);
+
+    static void EntryGroupCallback(AvahiServer *aServer,
+                                   AvahiSEntryGroup *aGroup, AvahiEntryGroupState aState,
+                                   AVAHI_GCC_UNUSED void *aUserData);
+    void EntryGroupCallback(AvahiServer *aServer, AvahiSEntryGroup *aGroup,
+                            AvahiEntryGroupState aState);
+
+    static void EntryGroupCallback(AvahiEntryGroup *aGroup, AvahiEntryGroupState aState,
+                                   AVAHI_GCC_UNUSED void *aUserData);
+    void EntryGroupCallback(AvahiEntryGroup *aGroup, AvahiEntryGroupState aState);
 
-private:
-    static int CreateService(AvahiServer *aServer);
-    static void EntryGroupCallback(AvahiServer *aServer, AvahiSEntryGroup *aGroup,
-                                   AvahiEntryGroupState aState, AVAHI_GCC_UNUSED void *aUserData);
     static void ServerCallback(AvahiServer *aServer, AvahiServerState aState,
                                AVAHI_GCC_UNUSED void *aUserData);
-    static void PeriodicallyPublish(AVAHI_GCC_UNUSED AvahiTimeout *aTimeout, void *aUserData);
-    static void Free(void);
-
-    static AvahiSEntryGroup *mGroup;
-    static AvahiSimplePoll  *mSimplePoll;
-    static AvahiServer      *mServer;
-    static const char       *mNetworkNameTxt;
-    static const char       *mExtPanIdTxt;
-    static const char       *mType;
-    static uint16_t          mPort;
-    static char             *mServiceName;
-    static bool              isStart;
+    void ServerCallback(AvahiServer *aServer, AvahiServerState aState);
+
+    static void ClientCallback(AvahiClient *aClient, AvahiClientState aState,
+                               AVAHI_GCC_UNUSED void *aUserData);
+    void ClientCallback(AvahiClient *aClient, AvahiClientState aState);
+
+    static void PeriodicallyPublish(AVAHI_GCC_UNUSED AvahiTimeout *aTimeout, AVAHI_GCC_UNUSED void *aUserData);
```

Comment:
Methods should include verbs.  `PeriodicallyPublish` does not tell me what this method actually does.